ambry-api/src/main/java/com/github/ambry/quota/QuotaManager.java

```diff
@@ -32,33 +32,58 @@
   void init() throws InstantiationException;
 
   /**
-   * Computes the overall boolean recommendation to throttle a request or not for all the types of request quotas supported.
+   * Computes the recommendation to throttle a request or not based only on usage exceeding quota.
    * This method does not charge the requestCost against the quota.
    * @param restRequest {@link RestRequest} object.
-   * @return ThrottlingRecommendation object that captures the overall recommendation.
+   * @return ThrottlingRecommendation object that captures the recommendation.
    */
   ThrottlingRecommendation getThrottleRecommendation(RestRequest restRequest);
 
   /**
    * Charges the requestCost against the quota for the specified restRequest and blobInfo.
+   * Atomically charges the requestCost against the quota for the QuotaResource in the specified restRequest and blobInfo if quota
+   * exceed is allowed.
```

Comment:
The comment is confusing. It first says " Atomically charges the requestCost against the quota .." and then says "This method does not charge the requestCost against the quota."

AFAICT, the first comment is true but is that the right thing to do? Do we always charge been though the throttling decision comes back as THROTTLE?